07.SubpassBaking/main.cpp

```diff
@@ -420,7 +423,11 @@ class SubpassBaking : public ApplicationBase
                         video::IGPUBuffer::SCreationParams scratchParams = {};
                         scratchParams.canUpdateSubRange = true;
                         scratchParams.usage = core::bitflag(video::IGPUBuffer::EUF_TRANSFER_DST_BIT) | video::IGPUBuffer::EUF_STORAGE_BUFFER_BIT;
-                        scratch = { 0ull,logicalDevice->createDeviceLocalGPUBufferOnDedMem(scratchParams,ppHandler->getMaxScratchSize()) };
+                        scratchParams.size = ppHandler->getMaxScratchSize();
+                        auto scratchBuff = logicalDevice->createBuffer(scratchParams);
+                        auto memReqs = scratchBuff->getMemoryReqs();
+                        logicalDevice->allocate(memReqs, scratchBuff.get());
```

Comment:
You need `memReqs.memoryTypeBits &=` I'll let you figure out the right hand side of it